declaring const c string causes c compiler error: cannot convert 'char [5]' to 'extern struct string'

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.8 18eee34.04aba8f
OS: windows, Microsoft Windows 11 Pro v22631 64-bit
Processor: 24 cpus, 64bit, little endian, 

getwd: D:\vbug2
vexe: C:\v\v.exe
vexe mtime: 2024-09-30 05:47:23

vroot: OK, value: C:\v
VMODULES: OK, value: C:\Users\info\.vmodules
VTMP: OK, value: C:\Users\info\AppData\Local\Temp\v_0

Git version: git version 2.42.0.windows.2
Git vroot status: 0.4.8-7-g04aba8f7-dirty
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,

operable program or batch file.


thirdparty/tcc status: thirdparty-windows-amd64 b425ac82

```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg main.v`
```v
module main

const msg = c'test'

fn main() {}
```

**What did you expect to see?**

no c compiler error


**What did you see instead?**
```
================== C compilation error (from tcc): ==============
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01J915VWT3FW80Q61TZVXRWFWW.tmp.c:7010: warning: implicit declaration of function 'tcc_backtrace'
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01J915VWT3FW80Q61TZVXRWFWW.tmp.c:13500: error: cannot convert 'char [5]' to 'extern struct string'
... (the original output was 3 lines long, and was truncated to 2 lines)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -5953,9 +5953,10 @@ fn (mut g Gen) const_decl(node ast.ConstDecl) {
 			}
 			ast.StringLiteral {
 				val := g.expr_string(field.expr)
+				typ := if field.expr.language == .c { 'char*' } else { 'string' }
 				g.global_const_defs[util.no_dots(field.name)] = GlobalConstDef{
 					mod:   field.mod
-					def:   'string ${const_name}; // a string literal, inited later'
+					def:   '${typ} ${const_name}; // a string literal, inited later'
 					init:  '\t${const_name} = ${val};'
 					order: -1
 				}

--- a/vlib/v/tests/consts/const_c_string_test.v
+++ b/vlib/v/tests/consts/const_c_string_test.v
@@ -1,0 +1,7 @@
+const msg = c'test'
+
+fn test_main() {
+	unsafe {
+		assert msg.vstring() == 'test'
+	}
+}

